src/c76_menuCAL.hpp

```diff
@@ -1232,27 +1249,36 @@ void printCalibrationSubmenu()
   else if (calState == ROLL_OFFSET_CALIBRATION)
   {
     auto angles = Gyro::getCurrentAngles();
-    sprintf(scratchBuffer, "R: -------------");
-    makeIndicator(scratchBuffer, angles.rollAngle - mount.getRollCalibrationAngle());
-    lcdMenu.printMenu(scratchBuffer);
+    float delta = accumulator.rollAngle - angles.rollAngle;
```

Comment:
I still think that this is not the place for the calculations, they should be abstracted in the gyro interface such that the menu code doesn't care about how the calculations work. The menu just wants the angles and the gyro takes care of how that happens, which reduces module coupling (https://en.wikipedia.org/wiki/Coupling_(computer_programming))

Reply:
I disagree. As a consumer of the class, I might not want it smoothed. I might want a different smoothing factor. I might be calling it irregularly and smoothing would be worse....

Reply:
Then it should be a parameter on gyro api. But here menu is changing actual behaviour while it should only trigger a feature without knowing the actual implementation.

Imagine we want to provide this as meade as well.

Reply:
> As a consumer of the class, I might not want it smoothed. I might want a different smoothing factor.

Simple, two functions then: `Gyro::getCurrentAnglesRaw(void)` and `Gyro::getCurrentAnglesFiltered(float smoothFactor)`

> I might be calling it irregularly and smoothing would be worse

That's up to the consumer of the class to use the interface correctly. You can fix that in the gyro interface with more complexity (threading, etc) but that's not needed for simple usage like this.